aten/src/THNN/generic/MultiLabelMarginCriterion.c

```diff
@@ -252,7 +252,7 @@ void THNN_(MultiLabelMarginCriterion_updateGradInput)(
   THTensor_(free)(input);
   THIndexTensor_(free)(target);
   THTensor_(free)(isTarget);
-  THTensor_(free)(gradOutput);
+  THTensor_(free)(gradInput);
```

Comment:
You shouldn't free gradInput -- I don't think you've made a copy of it.

Reply:
Haven't I now made a copy at line 196?

Reply:
Oops sorry I misspelled that!